### PR TITLE
Upgrade cmake build to grpc 162

### PR DESCRIPTION
**Pull request by wu-hui**
Note: we need to change the cocopods files for the test projects because newer version of grpc generate the pb.cc files that require updating the cocoapods configurations. It only impact how we run our tests through xcodebuild, and the actual SDK is not impacted.

There are other various small changes needed to make everything build, i have left comments to call them out. The majority of the changes are coming from generated pb.cc files as a result of protobufcpp upgrade.

### CODE REVIEW

File: Firestore/Example/ProtobufCpp.podspec [hunk 1]
```diff
@@ -33,7 +33,9 @@ Pod::Spec.new do |s|
   s.osx.deployment_target = '10.13'
   s.tvos.deployment_target = '12.0'
 
-  s.source_files = 'src/**/*.{h,cc,inc}'
+  s.source_files = 'src/**/*.{h,cc,inc}',
+                   # utf8_range is not needed, and is giving out build errors.
+                   'third_party/utf8_range/*.{h,cc,inc}'
```
Comment (ehsannas) [inline Firestore/Example/ProtobufCpp.podspec:38]: The comment is confusing to me. The comment says "utf8_range is not needed", and the next line adds `utf8_range` ot the `s.source_files`!

File: Firestore/Example/ProtobufCpp.podspec [hunk 2]
```diff
@@ -46,20 +48,38 @@ Pod::Spec.new do |s|
...
+  s.dependency 'abseil/container/btree', abseil_version
+  s.dependency 'abseil/container/flat_hash_map', abseil_version
+  s.dependency 'abseil/container/flat_hash_set', abseil_version
+  s.dependency 'abseil/log', abseil_version
+  s.dependency 'abseil/memory', abseil_version
+  s.dependency 'abseil/meta', abseil_version
+  s.dependency 'abseil/status', abseil_version
+  s.dependency 'abseil/strings/strings', abseil_version
+  s.dependency 'abseil/time', abseil_version
+  s.dependency 'abseil/types', abseil_version
```
Comment (ehsannas) [inline Firestore/Example/ProtobufCpp.podspec:73]: could we just do?
```
s.dependency 'abseil', '~> 1.20240116.1'
```

**Review by ehsannas — APPROVED** [review]
LGTM!

[needless to say] I didn't review any file that has
```
// Generated by the protocol buffer compiler. DO NOT EDIT!
```
at the top, and assumed the changes were auto-generated.

File: Firestore/Example/ProtobufCpp.podspec (hunk 1 above)
Comment (wu-hui) [reply Firestore/Example/ProtobufCpp.podspec:38]: Done.

**Comment by paulb777** [comment]
Should be good to merge now since we're on track to land SPM support for M145.

**Comment by paulb777** [comment]
I meant the comment above for #12398

**Pull request merged**